app/models/manageiq/providers/lenovo/physical_infra_manager/parsers/parser_dictionary_constants.rb

```diff
@@ -37,6 +37,29 @@ class ParserDictionaryConstants
     # and your values (string) corresponds to the attributes of the
     # source object who the value will be set as value of the key of result hash.
     # see +ManageIQ::Providers::Lenovo::Parse#parse+
+    SWITCH = {
+      :name          => 'name',
+      :uid_ems       => 'uuid',
+      :switch_uuid   => 'uuid',
+      :health_state  => 'overallHealthState',
+      :power_state   => 'powerState',
+      :product_name  => 'productName',
+      :part_number   => 'partNumber',
+      :serial_number => 'serialNumber',
+      :description   => 'description',
+      :manufacturer  => 'manufacturer',
+    }.freeze
+
+    SWITCH_PORT = {
+      :device_name => 'portName',
```

Comment:
Maybe _port_ should be used here instead of _port_name_. I notice that when I run this code against LXCA 2.0.0, the _port_name_ field is empty, resulting in the entry in the _guest_device_ table having a blank _device_name_. However, the _port_ field returned by LXCA contains the switch interface name such as "Ethernet1/2".

Reply:
Thank you for your email about that. As stated there, the [LXCA documentation](http://sysmgt.lenovofiles.com/help/index.jsp?topic=%2Fcom.lenovo.lxca_restapis.doc%2Frest_api_switches_uuid_ports_get.html&resultof="peerMacAddress"%20"peermacaddress"%20) presents the _port_ as the port index, and the name as _portName_. Also, I think we should maintain backward compatibility with previous LXCA versions, so we should still use the _portName_ . This is why I did chose that here. I prefer wait for that discussion before updating the code. What do you think about?

Reply:
Based on Ionut's latest response in the email thread, I believe that we want to use _port_ since he said it always contains the interface name.  Adding @rodneyhbrown7 to the conversation.

Reply:
Maybe it will be a good idea try to get the name of the port from `portName` and if it is empty, get from `port` once the official documentation says that `portName` really is the field for port name and `port` is "only" the index of the port. This way we provide a compatibility with LXCAs that provides such date in both fields.

Reply:
As we discussed in our meeting I will implement as suggested by @douglasgabriel. So we can have both. Thank you, @skovic, for finding that problem.